### PR TITLE
[nrf noup] boot/zephyr: add nrf54l15dk ext flash configs

### PR DESCRIPTION
**Pull request by nvlsianpu**
Copied configs from nrf54l15pdk.

### CODE REVIEW

File: boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp_ext_flash.conf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CONFIG_MULTITHREADING=y
```
Comment (nordicjm) [inline boot/zephyr/boards/nrf54l15dk_nrf54l15_cpuapp_ext_flash.conf:1]: *noup

**Review by nordicjm — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by michalek-no** [comment]
rebase

**Review by nordicjm — CHANGES_REQUESTED** [review]
erm, no, something amiss with your commits, 6 of them shouldn't be there

File: boot/zephyr/Kconfig [hunk 2]
```diff
@@ -202,7 +202,7 @@ config BOOT_SIGNATURE_TYPE_PURE_ALLOW
 
 choice BOOT_SIGNATURE_TYPE
 	prompt "Signature type"
-	default BOOT_SIGNATURE_TYPE_ED25519 if BOARD_NRF54L15PDK_NRF54L15_CPUAPP
+	default BOOT_SIGNATURE_TYPE_ED25519 if SOC_NRF54L15_CPUAPP
```
Comment (nordicjm) [inline boot/zephyr/Kconfig:205]: this part needs to be a fixup of https://github.com/nrfconnect/sdk-mcuboot/commit/3cdcdb32f254248e6d2cd87dce09597723782562 in separate commit

**Review by nordicjm — APPROVED** [review]

**Pull request merged**